docs/Logging/index.md

```diff
@@ -0,0 +1,77 @@
+| TRACE     | 0     | CF_Log.Trace    | Logs that contain the most detailed messages. These messages should never be enabled in a production environment.                                                                             |
+| DEBUG     | 1     | CF_Log.Debug    | Logs that should primarily contain information useful for debugging.                                                                                                                          |
+| INFO      | 2     | CF_Log.Info     | Logs that output information that may be relevant to the user.                                                                                                                                |
+| WARNING   | 3     | CF_Log.Warn     | Logs that highlight an abnormal action, but does cause the user to notice anything different.                                                                                                 |
+| ERROR     | 4     | CF_Log.Error    | Logs that highlight when the current flow of execution is stopped due to a failure. This should indicate if the current activity has a failure and will not result in the game shutting down. |
+| CRITICAL  | 5     | CF_Log.Critical | Logs that describe an unrecoverable event and will most likely lead to the shutdown of the game.                                                                                              |
+| NONE      | 6     |                 |                                                                                                                                                                                               |
+
+# Context Tracing
+
+Useful for when you have a crash and you just aren't sure where or what is causing it. Add these trace methods in every method. As a precautionary reminder, all outputs heavily impact performance. It is also recommended to place these trace function calls behind preprocessor defines and disabling them in production to improve performance, preventing filling up players drives and so if you ever need to find a new possible cause for crashing you can enable the define.
+
+## Example
+
+```csharp
+void SomeGlobalFunction()
+{
+	auto trace = CF_Trace_0("SomeGlobalFunction");
+	
+	SomeClass someClass = new SomeClass();
+	someClass.MethodOne(0.5);
+}
+
+class SomeClass
+{
+	ref CF_Trace m_SomeClassTrace = CF_Trace_Instance(this);
+	
+	void SomeClass()
+	{
+		auto trace = CF_Trace_0(this, "SomeClass");
+	}
+	
+	void MethodOne(float pDt)
+	{
+		auto trace = CF_Trace_1(this, "SomeFunction").Add(pDt);
+
+		MethodTwo("This is a test string", GetGame().GetPlayer().GetParent());
+	}
+
+	void MethodTwo(string message, IEntity parent)
+	{
+		auto trace = CF_Trace_2(this, "SomeFunction").Add(message).Add(parent);
+
+		MethodThree();
+	}
+
+	void MethodThree()
+	{
+		auto trace = CF_Trace_0(this, "MethodThree");
+	}
+}
+```
```

Comment:
I think its a great feature, but we need to make people understand it more easily.
This example alone does not tell me what to do, and I don't really understand it.

We need an explanation of each of the functions available. What does X do and when to use X for things like:
- CF_Trace_Instance
- CF_Trace_N
Also explain what CF_Trace_1 vs CF_Trace_2 is.

And you should mention that `auto trace = ` is required to be present.

Maybe when all of this is explained beforehand, this example makes more sense. I think it will probably not be enough content to justify splitting it into many pages. I think one "longer" page to explain the basics first and then some step-by-step examples from very basic to more advanced use cases might be the right call here.

Reply:
Rewrote and added mini documentation before the big example.